src/request/export/ExportWords.js

```diff
@@ -78,52 +92,61 @@ class ExportWords extends Nimiq.Observable {
-
-        TopLevelApi.focusPasswordBox();
     }
 
     /**
-     * @param {string} [password]
+     * @param {string} password
      */
     async _passwordSubmitted(password) {
         TopLevelApi.setLoading(true);
 
+        const passwordBuffer = password ? Utf8Tools.stringToUtf8ByteArray(password) : undefined;
         /** @type {Key?} */
         let key = null;
-
-        if (this._key) {
-            key = this._key;
-            password = this._password;
-        } else {
-            const passwordBuffer = password ? Utf8Tools.stringToUtf8ByteArray(password) : undefined;
-
-            try {
-                key = this._request.keyInfo.useLegacyStore
-                    ? await AccountStore.instance.get(
-                        this._request.keyInfo.defaultAddress.toUserFriendlyAddress(),
-                        /** @type {Uint8Array} */ (passwordBuffer),
-                    )
-                    : await KeyStore.instance.get(this._request.keyInfo.id, passwordBuffer);
-            } catch (e) {
-                if (e.message === 'Invalid key') {
-                    this._wordsPasswordBox.onPasswordIncorrect();
-                    TopLevelApi.setLoading(false);
-                    return;
-                }
-                this._reject(new Errors.CoreError(e));
+        try {
+            key = this._request.keyInfo.useLegacyStore
+                ? await AccountStore.instance.get(
+                    this._request.keyInfo.defaultAddress.toUserFriendlyAddress(),
+                    /** @type {Uint8Array} */ (passwordBuffer),
+                )
+                : await KeyStore.instance.get(this._request.keyInfo.id, passwordBuffer);
+        } catch (e) {
+            if (e.message === 'Invalid key') {
+                this._wordsPasswordBox.onPasswordIncorrect();
+                TopLevelApi.setLoading(false);
                 return;
             }
+            this._reject(new Errors.CoreError(e));
+            return;
+        }
 
-            if (!key) {
-                this._reject(new Errors.KeyNotFoundError());
-                return;
-            }
-            this.fire(ExportWords.Events.KEY_CHANGED, key, password);
-            this.setKey(key, password);
+        if (!key) {
+            this._reject(new Errors.KeyNotFoundError());
+            return;
+        }
+        this.fire(ExportWords.Events.KEY_CHANGED, key, password);
+        this._goToRecoveryWords(key);
+    }
+
+    /**
+     * @param {Key} key
+     * @private
+     */
+    _goToRecoveryWords(key) {
+        let words = [''];
+        if (key.secret instanceof Nimiq.PrivateKey) {
+            words = Nimiq.MnemonicUtils.entropyToLegacyMnemonic(key.secret.serialize());
+        } else if (key.secret instanceof Nimiq.Entropy) {
+            words = Nimiq.MnemonicUtils.entropyToMnemonic(key.secret);
+        } else {
+            this._reject(new Errors.KeyguardError('Unknown secret type'));
```

Comment:
Can you add the type to the error message?
```suggestion
            this._reject(new Errors.KeyguardError(`Unknown secret type: ${key.secret.type}`));
```